client/components/modals/FileUploadModal/ImagePreview.tsx

```diff
@@ -0,0 +1,41 @@
+import { Box } from '@rocket.chat/fuselage';
+import React, { useState, ReactElement } from 'react';
+
+import GenericPreview from './GenericPreview';
+import PreviewSkeleton from './PreviewSkeleton';
+
+type ImagePreviewProps = {
+	url: string;
+	file: File;
+};
+
+const ImagePreview = ({ url, file }: ImagePreviewProps): ReactElement => {
+	const [error, setError] = useState(false);
+	const [loading, setLoading] = useState(true);
+
+	const handleLoad = (): void => setLoading(false);
+	const handleError = (): void => {
+		setLoading(false);
+		setError(true);
+	};
+
+	if (error) {
+		return <GenericPreview file={file} />;
+	}
+
+	return (
+		<>
+			{loading && <PreviewSkeleton />}
+			<Box
```

Comment:
In my opinion, we should place the image as background to prevent distortion!
**Example:**
![image](https://user-images.githubusercontent.com/27704687/127944877-79dc0bed-4ce9-4ea9-95e6-99048f1d88f0.png)

Reply:
```suggestion
			{!loading && <Box
```